File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        int answer = 0;

        Set<Integer> set = new HashSet<>();
        for (int num: nums) {
            set.add(num);
        }
        
        for (int num: nums) {
            if (!set.contains(num - 1)) {
                int length = 0;
                while (set.contains(num + length)) {
                    length++;
                }
                
                answer = Math.max(answer, length);
            }
        }
        
        return answer;
    }
}

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.kt
class Solution {
    fun longestConsecutive(nums: IntArray): Int {
        if (nums.isEmpty()) {
            return 0
        }
        
        val setNums = mutableSetOf<Int>()
        
        for (num in nums) {
            setNums.add(num)
        }
        
        var answer = 1
        for (num in setNums) {
            if (!setNums.contains(num - 1)) {
                var currentNum = num
                var currentSteak = 1
                while (setNums.contains(currentNum + 1)) {
                    currentNum++
                    currentSteak++
                }
                
                answer = Math.max(answer, currentSteak)
            }
        }
        
        return answer
    }
}